src/Credentials/CredentialProvider.php

```diff
@@ -54,14 +54,16 @@ class CredentialProvider
     const ENV_SHARED_CREDENTIALS_FILE = 'AWS_SHARED_CREDENTIALS_FILE';
 
     /**
-     * Create a default credential provider that first checks for environment
-     * variables, then checks for the "default" profile in ~/.aws/credentials,
+     * Create a default credential provider that
+     * first checks for environment variables,
+     * then check for role-based auth in ~/.aws/config,
+     * then check for credential_process in the "default" profile in ~/.aws/credentials,
```

Comment:
Above this line (after role-based, before credential_process), please add one about the SSO- for example:

      * then checks for cached SSO credentials from the CLI

Reply:
Done. Actually, replaces existing line 59 to make the terminology line up. Thanks!

Reply:
There are 9 entries in the default provider, and only 8 listed in the documentation- now the assume role one is missing

Reply:
the default provider has the following entries in order after your changes:
1) env
2) web identity (assume role)
3) sso
4) credential_process from credential file
5) credentials from credentials ini file
6) credential_process from config file
7) credentials from config ini file
8) ECS
9) EC2

Your documentation lists them as:
1) checks for environment variables
2) checks for cached SSO credentials from the CLI
3) check for credential_process in the "default" profile in ~/.aws/credentials
4)  checks for the "default" profile in ~/.aws/credentials
5) then for credential_process in the "default profile" profile in ~/.aws/config
6) checks for "profile default" profile in ~/.aws/config (which is the default profile of AWS CLI)
7) tries to make a GET Request to fetch credentials if ECS environment variable is presented
8) finally checks for EC2 instance profile credentials.

Reply:
Oh, I see what you are saying. Yeah, the environment variable is both the original stuff, ACCESS_KEY and secret + the web identity env vars + mounted volumes. I can add that. Np.

Reply:
Done, great catch!